src/schedulers.ts

```diff
@@ -42,11 +41,19 @@ export class PreemptiveScheduler implements Scheduler {
           }
         } catch (e) {
           if (/Maximum call stack/.test(e.toString())) {
-            const stacks: es.CallExpression[] = []
+            const stacks: any = []
             for (let i = 1; i <= 3; i++) {
-              stacks.push(context.runtime.frames[i - 1].callExpression!)
-            }
-            context.errors.push(new MaximumStackLimitExceeded(context.runtime.nodes[0], stacks))
+              let currentFrame = context.runtime.frames[i - 1]
+              while (!currentFrame.callExpression) {
+                if (currentFrame.parent) {
+                  currentFrame = currentFrame.parent
+                } else {
+                  break
+                }
+              }
+              stacks.push(currentFrame.callExpression || {callee: "unknown", args: []})
```

Comment:
For this I wanted to make sure that it does not go all the way up to the global frame, since I check for the existence of the parent as well.

Reply:
I think in the case that it goes all the way up to the global frame, we should just give what frames we have to the error, and fix the display of errors to not assume that there are exactly 3 frames (needs to handle the cases where it can't find any frames, just 1, or just 2).